packages/labs/eslint-plugin/src/test/rules/attribute-names_test.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ruleTester} from './rule-tester.js';
+import {attributeNames} from '../../rules/attribute-names.js';
+
+ruleTester.run('attribute-names', attributeNames, {
+  valid: [
+    'class Foo {}',
+    `class Foo {
+      static get properties() {
+        return {
+          whateverCaseYouWant: {type: String}
+        };
+      }
+    }`,
+    `import {LitElement} from 'lit';
+     class Foo extends LitElement {
+       static get properties() {
+         return {
+           lowercase: {type: String}
+         };
+       }
+     }`,
+    `import {LitElement} from 'lit';
+     class Foo extends LitElement {
+       static get properties() {
+         return {
+           camelCase: {type: String, attribute: 'lowercase'}
+         };
+       }
+     }`,
+    `import {LitElement} from 'lit';
+     class Foo extends LitElement {
+        @property({ type: String })
+        lowercase = 'foo';
+      }`,
+    `import {LitElement} from 'lit';
+     class Foo extends LitElement {
+       @property({ type: String, attribute: 'lowercase' })
+       camelCase = 'foo';
+     }`,
+    `import {LitElement} from 'lit';
+     class Foo extends LitElement {
+       @property({ type: String, attribute: false })
+       camelCase = 'foo';
+     }`,
+  ],
+
+  invalid: [
+    {
+      code: `import {LitElement} from 'lit';
+        class Foo extends LitElement {
+          static get properties() {
+            return {
+              camelCase: {type: String}
+            };
+          }
+        }`,
+      errors: [
+        {
+          line: 5,
+          column: 15,
+          messageId: 'casedPropertyWithoutAttribute',
+        },
+      ],
+    },
+    {
+      code: `import {LitElement} from 'lit';
+        class Foo extends LitElement {
+          static get properties() {
+            return {
+              camelCase: {type: String, attribute: 'stillCamelCase'}
+            };
+          }
+        }`,
+      errors: [
+        {
+          line: 5,
+          column: 41,
+          messageId: 'casedAttribute',
+        },
+      ],
+    },
+    {
+      code: `import {LitElement} from 'lit';
+        import {property} from 'lit/decorators.js';
+        class Foo extends LitElement {
+          @property({ type: String })
+          camelCase = 'foo';
+        }`,
+      errors: [
+        {
+          line: 4,
+          column: 11,
+          messageId: 'casedPropertyWithoutAttribute',
+        },
+      ],
+    },
```

Comment:
Huh, I hadn't seen this rule before. Do we like this rule? Do we want to recommend it?

The casedAttribute message seems good, but the casedPropertyWithoutAttribute doesn't seem great to me.

Reply:
https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/attribute-names.md

I'm planning on porting all the existing rules to ease migration to the new version. I'm sure there is some overlap with Rune's rules, and some we might want to deprecate going forward. I agree that a lot of perfectly fine code may have a camelCased property name and be making the choice to use the lower-case / case insensitive attribute name and shouldn't be flagged. In those cases we may want additional configuration for this rule, and we may want a migration-oriented configuration and not include this in the recommended set.

I should open an RFC on how the plan for all of this.